src/lepton/jpgcoder.cc

```diff
@@ -1533,6 +1533,53 @@ void prep_for_new_file() {
 
 void concatenate_files(int fdint, int fdout);
 
+// ZBR: memory-caching child of ibytestream                      
+struct caching_ibytestream : public ibytestream 
+{
+    static std::vector<unsigned char> g_input_data;  // buffer holding entire contents of the current input file. We made it global in order to avoid memory reallocation for each next file
+    size_t starting_pos;  // initial position in the underlying input stream
+    size_t input_pos;  // current reading position in the input data
+    
+    caching_ibytestream (ibytestream &base_stream) : ibytestream(base_stream)
```

Comment:
`base_stream` will be copied here. Why can't `caching_ibytestream()` accept all `ibytestream` needs and pass them to it?